base_classes/NXcg_half_edge_data_structure.nxdl.xml

```diff
@@ -54,41 +54,68 @@
          Such a data structure can be used to efficiently circulate around faces
          and iterate over vertices of a planar graph.
     </doc>
-    <field name="dimensionality" type="NX_POSINT" units="NX_UNITLESS"/>
-    <field name="number_of_vertices" type="NX_POSINT" units="NX_UNITLESS"/>
-    <field name="number_of_faces" type="NX_POSINT" units="NX_UNITLESS"/>
-    <field name="number_of_half_edges" type="NX_POSINT" units="NX_UNITLESS"/>
+    <field name="dimensionality" type="NX_POSINT" units="NX_UNITLESS">
+        <doc>
+             Dimensionality of the primitives described.
+        </doc>
+    </field>
+    <!--resulting in a design similar to that of NXoff_geometry and the XDMF mixed primitive topology-->
+    <field name="number_of_vertices" type="NX_INT" units="NX_UNITLESS">
+        <doc>
+             Number of vertices for each face.
+             
+             Each entry represents the total number of vertices for that face,
+             irrespectively whether vertices are shared among faces or not.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="n_f"/>
+        </dimensions>
+    </field>
+    <field name="number_of_edges" type="NX_INT" units="NX_UNITLESS">
+        <doc>
+             Number of edges for each face.
+             
+             Each entry represents the total number of edges for that face,
+             irrespectively whether edges are shared across faces or not.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="n_e"/>
+        </dimensions>
+    </field>
+    <field name="number_of_faces" type="NX_INT" units="NX_UNITLESS">
+        <doc>
+             Number of faces of the primitives.
```

Comment:
Is this redundant? The shape of `number_of_vertices` gives this value.

Reply:
Yes, this is redundant, I removed it.